File: GARI_Umbria_2026-08-03/DotNet/AgronicaCoreStd/AgronicaCoreDTOStd/InData/Anagrafica/AppezzamentoXParcoMacchine.cs
using AgronicaCoreModelsSTD.anagrafiche;
using System;
using System.Collections.Generic;
using System.Text;

namespace InData.Anagrafica
{
    public class AppezzamentoXParcoMacchine
    {
        public string piva { get; set; }
        public int saCod { get; set; }
        public int appezza { get; set; }
        public int macCod { get; set; }
        public string classCode { get; set; }
        public IntervalloTemporale validity { get; set; }

        public AppezzamentoXParcoMacchine(
            string piva = "",
            int saCod = 0,
            int appezza = 0,
            int macCod = 0,
            string classcode = "",
            DateTime? startV = null,
            DateTime? endV = null
            )
        {
            this.piva = piva;
            this.saCod = saCod;
            this.appezza = appezza;
            this.macCod = macCod;
            this.classCode = classcode;
            this.validity = new IntervalloTemporale(
                startV.GetValueOrDefault(new DateTime(1900, 1, 1)),
                endV.GetValueOrDefault(new DateTime(2100, 12, 31))
                );
        }

        public AppezzamentoXParcoMacchine()
        {
            this.piva = "";
            this.saCod = 0;
            this.appezza = 0;
            this.macCod = 0;
            this.classCode = "";
            this.validity = new IntervalloTemporale(new DateTime(1900, 1, 1), new DateTime(2100, 12, 31));
        }
    }
}

File: GARI_Umbria_2026-08-03/GiasNG/src/app/Model/anagrafiche/appezzamento-x-parco-macchine.ts
import {AGRODATAFINE, AGRODATAINIZIO} from '../CostantiPersonalizzate';
import {IntervalloTemporale} from './IntervalloTemporale';

export class AppezzamentoXParcoMacchine {
  piva: string;
  saCod: number;
  appezza: number;
  macCod: number;
  classCode: string;
  validity: IntervalloTemporale;

  constructor(
    piva: string = '',
    saCod: number = 0,
    appezza: number = 0,
    macCod: number = 0,
    classCode: string = '',
    startValidity: Date = AGRODATAINIZIO,
    endValidity: Date = AGRODATAFINE
  ) {
    this.piva = piva;
    this.saCod = saCod;
    this.appezza = appezza;
    this.macCod = macCod;
    this.classCode = classCode;
    this.validity = new IntervalloTemporale(startValidity, endValidity);
  }
}
